examples/experimental/dagster-airlift/dagster_airlift/in_airflow/mark_as_migrating.py

```diff
@@ -25,21 +28,66 @@ def mark_as_dagster_migrating(
     caller_module = global_vars.get("__module__")
     suffix = f" in module `{caller_module}`" if caller_module else ""
     logger.debug(f"Searching for dags migrating to dagster{suffix}...")
-    num_dags = 0
-    for obj in global_vars.values():
-        if not isinstance(obj, DAG):
-            continue
-        dag: DAG = obj
-        logger.debug(f"Checking dag with id `{dag.dag_id}` for migration state.")
-        migration_state_for_dag = migration_state.get_migration_dict_for_dag(dag.dag_id)
-        if migration_state_for_dag is None:
-            logger.debug(
-                f"Dag with id `{dag.dag_id} hasn't been marked with migration state. Skipping..."
-            )
-        else:
-            dag.tags.append(json.dumps({"DAGSTER_MIGRATION_STATUS": migration_state_for_dag}))
-            logger.debug(
-                f"Dag with id `{dag.dag_id}` has been marked with migration state. Adding state to tags for dag."
+    dag_vars_to_mark = set()
+    task_vars_to_migrate = set()
+    all_dags_by_id = {}
+    for var, obj in global_vars.items():
+        if isinstance(obj, DAG):
+            dag: DAG = obj
+            if migration_state.dag_is_being_migrated(obj.dag_id):
+                logger.debug(f"Dag with id `{dag.dag_id}` has migration state.")
+                dag_vars_to_mark.add(var)
+            else:
+                logger.debug(
+                    f"Dag with id `{dag.dag_id} has no associated migration state. Skipping..."
+                )
+            all_dags_by_id[obj.dag_id] = obj
+        if isinstance(obj, BaseOperator) and migration_state.get_migration_state_for_task(
+            dag_id=obj.dag_id, task_id=obj.task_id
+        ):
+            task_vars_to_migrate.add(var)
+
+    for var in dag_vars_to_mark:
+        dag: DAG = global_vars[var]
+        logger.debug(f"Tagging dag {dag.dag_id} as migrating.")
+        dag.tags.append(
+            json.dumps(
+                {"DAGSTER_MIGRATION_STATUS": migration_state.get_migration_dict_for_dag(dag.dag_id)}
             )
-            num_dags += 1
-    logger.info(f"Marked {num_dags} dags as migrating to dagster{suffix}.")
+        )
+    logging.debug(f"Marked {len(dag_vars_to_mark)} dags as migrating to dagster via tag.")
+
+    for var in task_vars_to_migrate:
+        original_op: BaseOperator = global_vars[var]
```

Comment:
Confused what is going on here. The global namespace contains an operator?

Reply:
I am now realizing this is way overcomplicated and actually not correct. We should just be using the underlying task dictionary on the dag, and don't need to worry about operator globals at all (they aren't even guaranteed to exist).